common/persistence/elasticsearch/visibility_store.go

```diff
@@ -1032,3 +1032,15 @@ func cleanDSL(input string) string {
 	result := re.ReplaceAllString(input, `$2`)
 	return result
 }
+
+func convertClientError(message string, err error) error {
+	switch e := err.(type) {
+	case *elastic.Error:
+		switch e.Status {
+		case 400: // BadRequest
+			// Returning InvalidArgument error will prevent retry on a caller side.
+			return serviceerror.NewInvalidArgument(fmt.Sprintf("%s: %v", message, err))
```

Comment:
we should not see bad request from ES unless we have a bug
if user (our user) is issuing query which contains illegal argument, when we should return bad request from our end.